test/e2e/framework/cluster.go

```diff
@@ -0,0 +1,330 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/discovery"
+	cacheddiscovery "k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultUserAgent = "scylla-operator-e2e"
+	defaultQPS       = 20
+	defaultBurst     = 50
+)
+
+type Cluster struct {
+	name      string
+	namespace *corev1.Namespace
+
+	adminClientConfig *restclient.Config
+	clientConfig      *restclient.Config
+	username          string
+}
+
+func NewCluster(name string, restConfig *restclient.Config) *Cluster {
+	adminClientConfig := restclient.CopyConfig(restConfig)
+	adminClientConfig.UserAgent = defaultUserAgent
+	adminClientConfig.QPS = defaultQPS
+	adminClientConfig.Burst = defaultBurst
+
+	return &Cluster{
+		name: name,
+
+		adminClientConfig: adminClientConfig,
+		username:          "admin",
+	}
+}
+
+func (c *Cluster) Namespace() string {
+	return c.namespace.Name
+}
+
+func (c *Cluster) Username() string {
+	return c.username
+}
+
+func (c *Cluster) FieldManager() string {
+	h := sha512.Sum512([]byte(fmt.Sprintf("scylla-operator-e2e-%s", c.Namespace())))
+	return base64.StdEncoding.EncodeToString(h[:])
+}
+
+func (c *Cluster) ClientConfig() *restclient.Config {
+	return c.clientConfig
+}
+
+func (c *Cluster) AdminClientConfig() *restclient.Config {
+	return c.adminClientConfig
+}
+
+func (c *Cluster) DiscoveryClient() *discovery.DiscoveryClient {
+	client, err := discovery.NewDiscoveryClientForConfig(c.ClientConfig())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return client
+}
+
+func (c *Cluster) DynamicClient() dynamic.Interface {
+	client, err := dynamic.NewForConfig(c.ClientConfig())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return client
+}
+
+func (c *Cluster) DynamicAdminClient() dynamic.Interface {
+	client, err := dynamic.NewForConfig(c.AdminClientConfig())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return client
+}
+
+func (c *Cluster) KubeClient() *kubernetes.Clientset {
+	client, err := kubernetes.NewForConfig(c.ClientConfig())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return client
+}
+
+func (c *Cluster) KubeAdminClient() *kubernetes.Clientset {
+	client, err := kubernetes.NewForConfig(c.AdminClientConfig())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return client
+}
+
+func (c *Cluster) ScyllaClient() *scyllaclientset.Clientset {
+	client, err := scyllaclientset.NewForConfig(c.ClientConfig())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return client
+}
+
+func (c *Cluster) ScyllaAdminClient() *scyllaclientset.Clientset {
+	client, err := scyllaclientset.NewForConfig(c.AdminClientConfig())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return client
+}
+
+func (c *Cluster) CommonLabels() map[string]string {
+	return map[string]string{
+		"e2e":               "scylla-operator",
+		"framework-cluster": c.name,
```

Comment:
What's the point of labeling objects by cluster within the cluster?
Maybe we should just plumb in the framework name.

Reply:
done